### PR TITLE
Add assert that DoFCellAccessor::get/set_mg_dof_indices()

### PR DESCRIPTION
**Pull request by peterrum**

### CODE REVIEW

**Review by kronbichler — APPROVED** [review]

File: include/deal.II/dofs/dof_accessor.templates.h [hunk 1]
```diff
@@ -2600,6 +2600,10 @@ inline void
 DoFCellAccessor<dimension_, space_dimension_, level_dof_access>::
   get_mg_dof_indices(std::vector<types::global_dof_index> &dof_indices) const
 {
+  Assert(
+    this->dof_handler->mg_vertex_dofs.size() > 0,
+    ExcMessage(
+      "Multigrid DoF indices can be accessed if DoFHandler::distribute_mg_dofs() has been called!"));
```
Comment (tjhei) [inline include/deal.II/dofs/dof_accessor.templates.h:2606]:
```suggestion
      "Multigrid DoF indices can only be accessed after DoFHandler::distribute_mg_dofs() has been called!"));
```
(and below)

Comment (peterrum) [reply include/deal.II/dofs/dof_accessor.templates.h]: Changed!

**Review by kronbichler — APPROVED** [review]

File: include/deal.II/dofs/dof_accessor.templates.h [hunk 2]
```diff
@@ -2600,6 +2600,10 @@ inline void
 DoFCellAccessor<dimension_, space_dimension_, level_dof_access>::
   get_mg_dof_indices(std::vector<types::global_dof_index> &dof_indices) const
 {
+  Assert(
+    this->dof_handler->mg_vertex_dofs.size() > 0,
+    ExcMessage(
+      "Multigrid DoF indices can only be accessed after DoFHandler::distribute_mg_dofs() has been called!"));
```
Comment (kronbichler) [inline include/deal.II/dofs/dof_accessor.templates.h:2606]: Or wait, isn't this line longer than 80 characters and should better be broken into two?

Comment (peterrum) [reply include/deal.II/dofs/dof_accessor.templates.h:2606]: It's not like we are too consistent in this regard^^ But I'll change it tomorrow.

Comment (peterrum) [reply include/deal.II/dofs/dof_accessor.templates.h]: I have split the text up (and also other instances in that file that were too long).

**Review by kronbichler — APPROVED** [review]

**Pull request merged**